vault/expiration.go

```diff
@@ -114,6 +119,91 @@ func (c *Core) stopExpiration() error {
 	return nil
 }
 
+// Tidy cleans up stale leases which are associated with invalid tokens
+func (m *ExpirationManager) Tidy() error {
+	var tidyErrors *multierror.Error
+
+	// Create a cache to keep track of looked up tokens
+	tokenCache := make(map[string]string)
```

Comment:
The reason for doing this way is that, if we looked up a token and if that had turned out to be invalid, we need that information too. In comparison to the valid tokens, I assume that the number of invalid ones will be fairly less in number. While cleaning up, it is likely that we hit the same invalid token more than once. Caching information about a looked up invalid token might be of help.

I am okay to do the suggested changes. Just sharing my thoughts.